File: 0179-largest-number/0179-largest-number.ts
function largestNumber(nums: number[]): string {
    let allZeros = true;
    for(let i = 0; i < nums.length; i++){
        if(nums[i] != 0){
            allZeros = false;
            break
        }
    }
    if(allZeros){
        return "0"
    }
    return nums.sort((a, b) => (b.toString() + a.toString()).localeCompare(a.toString() + b.toString())).join("");
};

File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    string largestNumber(std::vector<int>& nums) {
        bool allZeros = true;
        for (int num : nums) {
            if (num != 0) {
                allZeros = false;
                break;
            }
        }
        if (allZeros) {
            return "0";
        }
        vector<string> strNums;
        for (int num : nums) {
            strNums.push_back(to_string(num));
        }
        sort(strNums.begin(), strNums.end(),
             [](const string a, const string b) { return b + a < a + b; });
        string result;
        for (const string s : strNums) {
            result += s;
        }
        return result;
    }
};